cfme/networks/network_router.py

```diff
@@ -42,7 +42,10 @@ def delete(self):
         """Deletes current cloud router"""
         view = navigate_to(self, 'Details')
         view.toolbar.configuration.item_select('Delete this Router', handle_alert=True)
-        view.flash.assert_success_message('Delete initiated for 1 Network Router.')
+        message = 'Delete initiated for 1 Network Router.'
+        wait_for(lambda: message in (m.read() for m in view.flash.messages),
```

Comment:
Suggestion, why not ```wait_for(lambda: view.flash.messages, fail_func=[], num_sec=10, delay=.1)``` and then assert ```view.flash.assert_success_message('Delete initiated for 1 Network Router.')```

Reply:
so, @jawatts suggestion only waits for there to be a flash message, not a matching flash message. Then once there is a flash message, it calls `assert_success_message` - which does two things. First it asserts there are no error messages, and second it asserts that the flash message given (Delete initiated...) is a success type message, and exists.  assert_message or assert_success_message handle getting the text from each `FlashMessage` widget generated by `FlashMessages`.